src/mscorlib/src/System/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -52,6 +52,18 @@ public static int SizeOf<T>()
         /// </summary>
         [NonVersionable]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static T As<T>(object o) where T : class
```

Comment:
The XML comment on this method no longer makes sense.

Reply:
Is there any value in making this instead be:
```C#
public static TTo As<TTo, TFrom>(TFrom o) where TTo : class, TFrom : class
```
?  That would help prevent it from being called with non-reference types.

Reply:
The non-reference type passed to this method will be boxed. It works fine.

Also, the methods in this type should match the methods in public Unsafe, and we have shipped that one with the `object` signature.

> The XML comment on this method no longer makes sense.

Copy the comment from the public Unsafe in corefx.

Reply:
> The non-reference type passed to this method will be boxed. It works fine.

Yeah, I wasn't concerned with it not working, but rather discouraging such use.  But if we've already shipped the API like this in the public version, then it makes sense to keep it like this.